src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        Figure figure;
+        GeometricFigure geometricFigure = new GeometricFigureSupplier().getRandomFigure();
```

Comment:
GeometricFigureSupplier, ColorSupplier and Random should be class-level variables